lib/Conversion/HandshakeToFIRRTL/HandshakeToFIRRTL.cpp

```diff
@@ -32,10 +32,46 @@ using ValueVectorList = std::vector<ValueVector>;
 // Utils
 //===----------------------------------------------------------------------===//
 
-/// Build a FIRRTL bundle type (with data, valid, and ready subfields) given the
-/// type of the data subfield.
-static FIRRTLType buildBundleType(FIRRTLType dataType, bool isFlip,
-                                  MLIRContext *context) {
+/// Get the corresponding FIRRTL type given the built-in data type. Current
+/// supported data types are integer (signed, unsigned, and signless), index,
+/// and none.
+static FIRRTLType getFIRRTLType(Type type) {
+  MLIRContext *context = type.getContext();
+  return TypeSwitch<Type, FIRRTLType>(type)
+      .Case<IntegerType>([&](IntegerType integerType) {
+        unsigned width = integerType.getWidth();
+
+        switch (integerType.getSignedness()) {
+        case IntegerType::Signed:
+          return FIRRTLType(SIntType::get(context, width));
```

Comment:
I don't think you need the casts here, this should be enough:

`return SIntType::get(context, width);`